src/theory/quantifiers/ceg_t_instantiator.cpp

```diff
@@ -934,19 +945,59 @@ Node BvInstantiator::hasProcessAssertion(CegInstantiator* ci, SolvedForm& sf,
         Assert(!sm.isNull() && sm.isConst());
         Node tm = ci->getModelValue(t);
         Assert(!tm.isNull() && tm.isConst());
+        Node ret;
         if (sm != tm) {
           Node slack =
               Rewriter::rewrite(nm->mkNode(kind::BITVECTOR_SUB, sm, tm));
           Assert(slack.isConst());
           // remember the slack value for the asserted literal
           d_alit_to_model_slack[lit] = slack;
-          Node ret = nm->mkNode(kind::EQUAL, s,
-                                nm->mkNode(kind::BITVECTOR_PLUS, t, slack));
+          ret = nm->mkNode(kind::EQUAL, s,
+                           nm->mkNode(kind::BITVECTOR_PLUS, t, slack));
           Trace("cegqi-bv") << "Process " << lit << " as " << ret
                             << ", slack is " << slack << std::endl;
-          return ret;
+        }else{
+          ret = s.eqNode(t);          
+          Trace("cegqi-bv") << "Process " << lit << " as " << ret << std::endl;
         }
+        return ret;
+      }
+    } else {
+      // otherwise, we optimistically solve for the boundary point of an inequality
+      if (!pol) {
+        if (k == BITVECTOR_ULT) {
+          k = BITVECTOR_UGE;
+        } else if (k == BITVECTOR_ULE) {
+          k = BITVECTOR_UGT;
+        } else if (k == BITVECTOR_SLT) {
+          k = BITVECTOR_SGE;
+        } else {
+          Assert(k == BITVECTOR_SLE);
+          k = BITVECTOR_SGT;
+        }
```

Comment:
SLT and SLE should be rewritten to ULT and ULE. Maybe Assert that (k == BITVECTOR_ULT && k == BITVECTOR_ULE) to make sure we don't miss anything and remove the SLT and SLE cases?